File: src/com/esprit/entities/sponsor.java
package com.esprit.entities;

import java.util.Objects;

/**
 *
 * @author oumayma
 */
public class sponsor {

    private int id;
    public String nomS;
    private String logoS;

    public sponsor() {
    }

    public sponsor(int id) {
        this.id = id;
    }

    public sponsor(int id, String nomS, String logoS) {
        this.id = id;
        this.nomS = nomS;
        this.logoS = logoS;
    }

    public sponsor(String nomS, String logoS) {
        this.nomS = nomS;
        this.logoS = logoS;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getnomS() {
        return nomS;
    }

    public void setnoms(String nomS) {
        this.nomS = nomS;
    }

    public String getlogoS() {
        return logoS;
    }

    public void setlogoS(String logoS) {
        this.logoS = logoS;
    }

    @Override
    public String toString() {
        return "serviceclub {" + "id=" + id + ", nomS=" + nomS + ", logoS=" + logoS + '}';
    }

}

File: src/Entity/Sponsor.php
<?php

namespace App\Entity;

use App\Repository\SponsorRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=SponsorRepository::class)
 */
class Sponsor
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $nomS;

    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     */
    private $logoS;

    /**
     * @ORM\OneToMany(targetEntity=Club::class, mappedBy="sponsor", orphanRemoval=true)
     */
    private $clubs;

    public function __construct()
    {
        $this->clubs = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomS(): ?string
    {
        return $this->nomS;
    }

    public function setNomS(string $nomS): self
    {
        $this->nomS = $nomS;

        return $this;
    }

    public function getLogoS(): ?string
    {
        return $this->logoS;
    }

    public function setLogoS(?string $logoS): self
    {
        $this->logoS = $logoS;

        return $this;
    }

    /**
     * @return Collection|Club[]
     */
    public function getClubs(): Collection
    {
        return $this->clubs;
    }

    public function addClub(Club $club): self
    {
        if (!$this->clubs->contains($club)) {
            $this->clubs[] = $club;
            $club->setSponsor($this);
        }

        return $this;
    }

    public function removeClub(Club $club): self
    {
        if ($this->clubs->removeElement($club)) {
            // set the owning side to null (unless already changed)
            if ($club->getSponsor() === $this) {
                $club->setSponsor(null);
            }
        }

        return $this;
    }

    public function __toString() {
        return $this->nomS;
    }
}
